dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -283,6 +283,11 @@ class dpnp_trapz_c_kernel;
 template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
 void dpnp_trapz_c(const void* array1_in, const void* array2_in, void* result1, double dx, size_t array1_size, size_t array2_size)
 {
+    if ((array1_in == nullptr) || (array2_in == nullptr && array1_size > 1))
```

Comment:
The condition is `True` if `array1 = np.array([1, 2, 3]); array2 = None`, but we don't need to return in this case. Isn't it?